File: tinyredis-core/src/main/java/com/github/cjqcn/tinyredis/core/db/Dict.java
package com.github.cjqcn.tinyredis.core.db;

public interface Dict<K, V> {

    V get(K key);

    void set(K key, V value);

    V setnx(K key, V value);

    boolean exist(K key);

    V remove(K key);


}

File: tinyredis-core/src/main/java/com/github/cjqcn/tinyredis/core/db/Dict.kt
package com.github.cjqcn.tinyredis.core.db

import com.github.cjqcn.tinyredis.core.struct.RedisObject

interface Dict {
    fun get(key: RedisObject): RedisObject?
    fun set(key: RedisObject, value: RedisObject): Boolean
    fun del(key: RedisObject): Boolean
}